backend/src/Controllers/castVoteController.ts

```diff
@@ -0,0 +1,187 @@
+import { Election, electionValidation } from "../../../domain_model/Election";
+import { ElectionRoll, ElectionRollState } from "../../../domain_model/ElectionRoll";
+import { Ballot, Ballot as BallotType, ballotValidation } from '../../../domain_model/Ballot';
+import { IRequest } from "../IRequest";
+import ServiceLocator from "../ServiceLocator";
+import Logger from "../Services/Logging/Logger";
+import { BadRequest, InternalServerError, Unauthorized } from "@curveball/http-errors";
+import { ILoggingContext } from "../Services/Logging/ILogger";
+import {expectUserFromRequest, expectValidElectionFromRequest, catchAndRespondError} from "./controllerUtils";
+import { randomUUID } from "crypto";
+
+const ElectionsModel = ServiceLocator.electionsDb();
+const ElectionRollModel = ServiceLocator.electionRollDb();
+const BallotModel = ServiceLocator.ballotsDb();
+const CastVoteStore = ServiceLocator.castVoteStore();
+
+
+async function castVoteController(req: IRequest, res: any, next: any) {
+    Logger.info(req, "Cast Vote Controller");
+
+    const user = expectUserFromRequest(req);
```

Comment:
User info would already be available in req.user. We get the user data if available on every request in app.ts 
`app.use(authController.getUser)`

We also don't need to assert this with every vote, just when the voter id is set to email.